lib/Controller/KeyController.php

```diff
@@ -205,6 +205,26 @@ public function createPublicKey(string $csr): DataResponse {
 		return new DataResponse(['public-key' => $publicKey]);
 	}
 
+	/**
+	 * Set public key
+	 *
+	 * @NoAdminRequired
+	 * @E2ERestrictUserAgent
+	 * @throws OCSBadRequestException
+	 */
+	public function setPublicKey(string $publicKey): DataResponse {
+		try {
+			$this->keyStorage->setPublicKey($publicKey, $this->userId);
+		} catch (KeyExistsException $e) {
+			return new DataResponse([], Http::STATUS_CONFLICT);
+		} catch (Exception $e) {
+			$this->logger->critical($e->getMessage(), ['exception' => $e, 'app' => $this->appName]);
```

Comment:
Error is enough, and `'exception' => $e` will contain the original exception message

```suggestion
			$this->logger->error("Fail to set user public key", ['exception' => $e, 'app' => $this->appName]);
```

Reply:
done